packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -210,9 +210,10 @@ class AnalyzeCommand extends FlutterCommand {
       return false;
     }
 
-    // Don't run pub if asking for android analysis.
+    // Android analyze needs to process resource, i.e. evaluating build
```

Comment:
Why do build settings or assets require pub to be run?

Reply:
I am not too familiar with android internal, but i need to run processResource gradle task to get the finalized andriodManifest, and that will cause flutter app to build.

Reply:
My guess is that part of pubspec.yaml assets like font or image will end up in the resource in the android project

Reply:
> i need to run processResource gradle task to get the finalized andriodManifest, and that will cause flutter app to build.

Ok, this sounds reasonable :)